test/users.js

```diff
@@ -223,4 +245,103 @@ describe('users test', () => {
         .then(() => done());
     });
   });
+  describe('/users/logout POST', () => {
+    it('Should successful because user logout correctly', done => {
+      return successfulLogin().then(res => {
+        res.should.have.status(201);
+        return sessionService
+          .getCount(res.body.email)
+          .then(count => {
+            count.should.to.equal(1);
+          })
+          .then(() => {
+            return chai
+              .request(server)
+              .post('/users/logout')
+              .send({ email: res.body.email })
+              .set(sessionManager.HEADER_NAME, res.headers[sessionManager.HEADER_NAME])
+              .then(res2 => {
+                return sessionService.getCount(res.body.email).then(count2 => {
+                  count2.should.to.equal(0);
+                });
+              })
+              .then(() => done());
+          });
+      });
+    });
+    it('Should fail because of invalid token', done => {
+      return successfulLogin().then(res => {
+        res.should.have.status(201);
+        return sessionService.getCount(res.body.email).then(count => {
+          count.should.to.equal(1);
+          setTimeout(() => {
+            return chai
+              .request(server)
+              .post('/users/logout')
+              .send({ email: res.body.email })
+              .set(sessionManager.HEADER_NAME, res.headers[sessionManager.HEADER_NAME])
+              .catch(err => {
+                err.response.should.have.status(401);
+                return sessionService.getCount(res.body.email).then(count2 => {
+                  count2.should.to.equal(0);
+                });
+              })
+              .then(() => done());
+          }, 1000);
+        });
+      });
+    });
+  });
+
+  describe('/users/logout/all POST', () => {
+    it('Should fail because of invalid token', done => {
+      return successfulLogin().then(res => {
+        res.should.have.status(201);
```

Comment:
is the second login necessary here?

Reply:
Yes. I want to see the behavior with multiple sessions.

Reply:
but you included this in all the subsequent tests, didnt you check this behaviour specifically in a previous one??